### PR TITLE
GeneralizedAlpha for 2nd order ODE 

### PR DESCRIPTION
**Pull request by shagun751**
Added GeneralizedAlpha methods for 2nd order ODE in GeneralizedAlpha.jl. 
Added the test in ODESolversTests.jl. 
The method is used in solving the Generalised Wave Continuity Equation.

### CODE REVIEW

File: src/ODEs/ODETools/GeneralizedAlpha.jl [hunk 1]
```diff
@@ -47,7 +47,60 @@ function solve_step!(
...
+  t0::Real,
+  cache) # -> (uF,tF)
+
+  dt = solver.dt
+  ρ∞ = solver.ρ∞
+  αf = ρ∞/(ρ∞ + 1.0)
+  αm = (2*ρ∞ - 1.0)/(ρ∞ + 1.0)
+  γ = 0.5 - αm + αf
+  β = 0.25*((1.0 - αm + αf)^2)
+  tαf = t0 + (1.0-αf)*dt
```
Comment (oriolcg) [inline src/ODEs/ODETools/GeneralizedAlpha.jl:65]: In the GeneralizedAlpha for 1st order  the parameters are defined as follows:
```
αf = 1.0/(1.0 + ρ∞)
  αm = 0.5 * (3-ρ∞) / (1+ρ∞)
  γ = 0.5 + αm - αf
  tαf = t0+αf*dt
  ```
  We should keep the same definitions.

File: src/ODEs/ODETools/GeneralizedAlpha.jl [hunk 2]
```diff
@@ -47,7 +47,60 @@ function solve_step!(
...
+    αf / (1.0 - αf) * u0
+
+  v1 = γ / (β*dt) * (u1 - u0) - 
+    (γ - β) / β * v0 - 
+    (γ - 2.0 * β) / (2.0 * β) * dt * a0
+  
+  a1 = 1.0 / (β * dt * dt) * (u1 - u0) - 
+    1.0 / (β * dt) * v0 - 
+    (1.0 - 2.0 * β) / (2.0 * β) * a0
+
```
[the remark this answers is not in the document]
Comment (oriolcg) [reply src/ODEs/ODETools/GeneralizedAlpha.jl:92]: In literature the coefficients are different for 1st and 2nd order. We keep it as they are now to be consistent with notation in literature

**Comment by oriolcg** [comment]
No need for the requested changes as the formulation is the same as in literature references. @santiagobadia @fverdugo could you review this PR?

**Comment by fverdugo** [comment]
Hi @oriolcg for me it is OK to merge this if it looks good to you.

**Comment by oriolcg** [comment]
Yes @fverdugo , you can go ahead

**Pull request merged**